### PR TITLE
Inducer Buffs: superinducer, emagging act and stunning people with inducers

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

**Super Inducer**
The Super Inducer is an upgraded inducer, similar to the one CE starts with. It's twice as fast but shocks you if you're not insulated/wearing insuls.

**Emagged Inducer**
Emagging an inducer adds a new functinality; recharging energy weapons. Should be an option for traitors who steal eGuns but cannot recharge them.

**Inducer Stun**
An unemagged inducer will only cause people to jitter and be confused (like ion guns), but a emagged one will shock, damaging and stunning the suckers, like a defib.

## Why It's Good For The Game

Inducers are based, but generally go underused because they are super slow, with no potential to upgrade.

Also, engineering traitors don't really have a proper direct stun compared to science, medical or even service.

## Changelog
:cl:
add: super inducer item and design to science/engineering lathe
add: you can use emagged inducers to shock people and recharge energy guns
add: You can overcarge machines with emagged inducers to damage them.
/:cl:

### CODE REVIEW

**Comment by Moccha-Bee** [comment]
good pr 👍

**Comment by VictorVow** [comment]
Stun combat is deprecated, swap it for a knockdown.

**Comment by Blueturbo47** [comment]
Inducer bundle? Emmaged inducer with cells for it? maybe?

**Comment by TeomanTheGreat** [comment]
Ah yes a traitor engi buff good work!

**Comment by ghost** [comment]
Hardstuns are incredibly unfun to fight against which is the reason they were largely removed, change it to a knockdown.

**Comment by eeSPee** [comment]
Does victim.electrocute_act hardstun or just knockdown?

**Comment by ghost** [comment]
> Does victim.electrocute_act hardstun or just knockdown?

I mean.. You could just test it.

**Comment by eeSPee** [comment]
Nevermind; looked up the function.

it DOES paralyze if stun is  TRUE.

**Comment by St0rmC4st3r** [comment]
>Inducer Stun
An unemagged inducer will only cause people to jitter and be confused (like ion guns), but a emagged one will shock, damaging and stunning the suckers, like a defib.

This is not how EM induction works.

**Review by zeskorion — CHANGES_REQUESTED** [review]
This is an electromagnet, which induces a current in another device. It's not a magic zappy stick

Perhaps give it a magnet function instead

**Comment by ZombiesVsPlants** [comment]
Telekinesis on items that have iron in them pog

**Comment by eeSPee** [comment]
> This is an electromagnet, which induces a current in another device. It's not a magic zappy stick
> 
> Perhaps give it a magnet function instead

Like, pulling metallic items towards yourself?

**Comment by utilbird** [comment]
Could make it act like a ghetto induction furnace and use it to burn silicons, damage machines ranged shutdown of cameras etc.

**Comment by eeSPee** [comment]
> Could make it act like a ghetto induction furnace and use it to burn silicons, damage machines ranged shutdown of cameras etc.

I like that.

**Comment by Isy232** [comment]
@eeSPee You're missing the opportunity to make it a ghetto railgun.

Have it loaded with metal rods and it shoots them out dealing meh damage with high embed chance or something.

Or I dunno add a new tator engi item.

Or have it do more damage depending on how much energy you put in it.

**Comment by eeSPee** [comment]
> Have it loaded with metal rods and it shoots them out dealing meh damage with high embed chance or something.

But the inducer doesn't have a slot to load iron rods. A specially modified inducer taht does that, mmmm.

> Or have it do more damage depending on how much energy you put in it.

That was my initial idea.

I'm thinking of atomizing this into two separate PRS: superinducer and inducer as a knockdown weapon

File: code/game/objects/items/inducer.dm [hunk 1]
```diff
@@ -97,24 +97,37 @@
...
 		recharging = TRUE
 	var/obj/item/stock_parts/cell/C = A.get_cell()
 	var/obj/O
 	var/coefficient = 1
-	if(istype(A, /obj/item/gun/energy))
+	if(istype(A, /obj/item/gun/energy) && !(obj_flags & EMAGGED))
 		to_chat(user,"Error unable to interface with device")
 		return FALSE
+	else
+		coefficient = 1/5
```
Comment (PowerfulBacon) [inline code/game/objects/items/inducer.dm]: This is going to set coefficient = 1/5 for everything

File: code/game/objects/items/inducer.dm [hunk 2]
```diff
@@ -131,13 +144,36 @@
...
-	if(user.a_intent == INTENT_HARM)
-		return ..()
-
 	if(cantbeused(user))
 		return
+		
+	/*if(user.a_intent == INTENT_HARM)	
+		if (istype(M, /mob/living/carbon))
+			do_harm(M,user)
+		return ..()*/
```
Comment (PowerfulBacon) [inline code/game/objects/items/inducer.dm]: Commented out code

File: code/game/objects/items/inducer.dm [hunk 3]
```diff
@@ -131,13 +144,36 @@
 		return TRUE
 	recharging = FALSE
 
+/obj/item/inducer/proc/do_harm(mob/living/carbon/victim, mob/living/user)	//maybe a good idea to leave this out?
```
Comment (PowerfulBacon) [inline code/game/objects/items/inducer.dm:155]: Is this even called from anywhere? If it's just commented out code, might as well remove it

File: code/game/objects/items/inducer.dm (hunk 1 above)
Comment (eeSPee) [reply code/game/objects/items/inducer.dm]: fixed

File: code/game/objects/items/inducer.dm (hunk 2 above)
Comment (eeSPee) [reply code/game/objects/items/inducer.dm]: Revised this section.

File: code/game/objects/items/inducer.dm (hunk 3 above)
Comment (eeSPee) [reply code/game/objects/items/inducer.dm:155]: I was considering removing it altogether, but changed my mind.

**Comment by eeSPee** [comment]
I tested it and everything works as intended.

File: code/game/objects/items/inducer.dm [hunk 4]
```diff
@@ -105,24 +105,41 @@
 	var/obj/O
 	var/coefficient = 1
 	if(istype(A, /obj/item/gun/energy))
-		to_chat(user,"Error unable to interface with device")
-		return FALSE
+		if (obj_flags & EMAGGED)		
+			coefficient = 1/5
```
Comment (ike709) [inline code/game/objects/items/inducer.dm]:
```suggestion
			coefficient = 0.2
```

Imagine doing division for no reason.

**Review by ike709 — CHANGES_REQUESTED** [review]
Lots of bad code.

**Comment by eeSPee** [comment]
Noted; will take a look over code again

File: code/game/objects/items/inducer.dm [hunk 5]
```diff
@@ -105,39 +105,72 @@
...
+				else
+					keep_looping = FALSE
+					
+				if (keep_looping)
 					O.update_icon()
+					do_sparks(1, FALSE, A)
+					if (powertransfer>1000)
+						var/mob/living/carbon/human = user
+						if (human.electrocute_act( (powertransfer-1000)/400,human,stun = TRUE))
+							keep_looping = FALSE
```
Comment (PowerfulBacon) [inline code/game/objects/items/inducer.dm:146]: Can't you just break instead?

Comment (eeSPee) [reply code/game/objects/items/inducer.dm:146]: And while (TRUE)?

Comment (eeSPee) [reply code/game/objects/items/inducer.dm:146]: Ok found a workaround

**Comment by ike709** [comment]
Code is still pretty bad, I suggest smaller PRs in the future until you're a better coder.

**Pull request closed without merging**